src/tiled/filesystemmodel.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * filesystemmodel.h
+ * Copyright 2015, Mamed Ibrahimov <ibramlab@gmail.com>
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef FILESYSTEMMODEL_H
+#define FILESYSTEMMODEL_H
+
+#include <QFileSystemModel>
+
+namespace Tiled {
+namespace Internal {
+
+/**
+ * Class represents the file system model with disabled dragging of directories.
+ */
+class FileSystemModel: public QFileSystemModel
```

Comment:
Rather than putting this in its own set of files, I'd just have this class in `mapsdock.cpp`. There is no need to change `mapsdock.h` even, since the variable can stay a `QFileSystemModel*`.